torch/csrc/jit/pybind_utils.h

```diff
@@ -76,24 +76,85 @@ inline void findErrorInKwargs(const FunctionSchema& schema, py::kwargs kwargs) {
 }
 } // namespace detail
 
-inline IValue toIValue(py::handle input) {
+using tracer::TypedStack;
+struct TypedIValue : public std::pair<IValue, TypePtr> {
+  using pair::pair;
+
+  IValue& ivalue() {
```

Comment:
There's another PR out right now that adds a NamedIValue with type, name, and value, so a follow maybe to subclass / consolidate on API

Reply:
Oh, OK. I was lacking that context. Happy to merge to a single C++ type however is convenient.

Reply:
Actually I think it's fine. From looking at the comments from NamedIValue, it's on the way to being tied to a Module, which doesn't really fit in here. 
```
// Currently this is internally implemented as a pointer, but when
// modules become first-class this will be a pair of  <module_ivalue, slot_number>

```